### DISCUSSION TITLE
Steam API Integration

### DISCUSSION THREAD

**Discussion by Genevieve-Nurse** [nesbox/TIC-80#2499] [Q&A] [answered]
Hi,

I'm currently gearing up to launch a TIC-80 game on Steam. It has an in-game achievements system, and I'm interested in hooking it up to the Steam Achievements system, but I don't know whether this is possible in TIC-80.

Is there a way of integrating the Steam API into a TIC-80 game? If not, would it be possible to add this in the future? It's more of a nice-to-have than anything, but I am slightly paranoid about the possibility of players getting annoyed that there are in-game achievements but no Steam Achievements.

**Comment by SolarLune** [comment]
Would it be possible to include / use bindings to a Steam API library for the language you're using (i.e. Luasteam for Lua)?

**Reply by Genevieve-Nurse** [reply]
I have no idea. I'm using Lua, and I wasn't aware that there was a Steam API library for it. I'll have a search around and see if there's a way I can include it, though if it's anything like trying to inject the Newgrounds API into a PICO-8 game, I'm probably going to need to be walked through it step-by-step. We'll see how things go.

**Reply by Genevieve-Nurse** [reply]
Case in point, I am currently struggling my way through the luasteam documentation, along with [borbware's guide to multi-file TIC-80 projects](https://borbware.github.io/tic80-basics/tic80-advanced/multi-file-projects-slides.html), and I'm getting a "too many C levels (limit is 200)" error and no idea why. This should be far, far simpler than I'm actually finding it.

**Comment by Genevieve-Nurse** [comment, marked as answer]
After doing some more digging, I'm afraid it doesn't seem like this is possible, at least in Lua. Luasteam needs to be instantiated via

`local Steam = require('luasteam')`
`Steam.init()`

...but in TIC-80's implementation of Lua, `require()` returns a boolean, not an object. So I can theoretically include the API, but there doesn't seem to be any way of actually accessing its contents.

Given that `require()` isn't supported anyway and may be removed in future (https://github.com/nesbox/TIC-80/issues/2426), it seems the answer to this is a resounding **no**: Steam API integration isn't possible, at least via Lua, and there are no plans to make it possible. Disappointing, but not too surprising.
